src/codegen/codegen_cuda.cpp

```diff
@@ -312,7 +313,7 @@ class CodeGen_CUDA::DeviceFunctionCollector : public IRVisitor {
       taco_iassert(var) << "Outputs must be vars in codegen";
       taco_iassert(scopeMap.count(var) == 0) <<
                                              "Duplicate output found in codegen";
-
+      output_tensor = var->name; // Isn't there only one output?
```

Comment:
The code generator was designed with the thought that we might eventually support kernels with multiple outputs, though that's not something we've actually fully implemented. It's probably fine to assume for now that there's only one output, though you should probably add an assertion to check for that (e.g., `taco_iassert(outputs.size() == 1)`) .

Reply:
I see. Thanks for explanation. I know understand why the code looks like that. I will add an assertion to check there is only one output.

Reply:
Thanks for your time to take a look at the parts I modified!